### ISSUE TITLE
Email email window is shown altough there is a content

### ISSUE THREAD

**Issue by thomascube** [thomascube/roundcube-trac-import#88]
**Reported by roland news on 4 Jun 2007 21:07 UTC**

Some of my emails are shown without content. The email seems to be empty altough there is a content in the email. I can only access the attachements but not th text.

The email can be shown without any probems in Outlook (using IMAP) and also with ilohamail (webmail), also using IMAP.

I dont know the difference to other mails... (see also screenshot)

I use Lumisoft mailserver
http://www.lumisoft.ee/lswww/ENG/Products/Mail_Server/mail_index_eng.aspx?type=info

Mailserver and Roundcube runs on Windows2k3

Return-Path: <xxx.xxx@xx.at>
X-Flags: 1001
Delivered-To: GMX delivery to roland.news@gmx.at
Received: (qmail invoked by alias); 15 May 2007 13:33:12 -0000
Received: from xlspmp03.mdcs.at (EHLO xlspmp03.mdcs.at) [ by mx0.gmx.net (mx097) with SMTP; 15 May 2007 15:33:12 +0200
Received: from localhost (localhost.localdomain [127.0.0.1](193.110.28.55]))	by xlspmp03.mdcs.at (Postfix) with ESMTP id D3FCB2EE960	for <roland.news@gmx.at>; Tue, 15 May 2007 15:33:11 +0200 (CEST)
X-Virus-Scanned: amavisd-new at mdcs.at
Received: from xlspmp03.mdcs.at ([localhost (xlspmp03.mdcs.at [127.0.0.1](127.0.0.1])	by)) (amavisd-new, port 10025)	with ESMTP id TLR4HS2msRf7 for <roland.news@gmx.at>;	Tue, 15 May 2007 15:33:11 +0200 (CEST)
Received: from mx02.mdcs.at (atrbgmdcs2.wien.rbgat.net [10.14.33.90])	by xlspmp03.mdcs.at (Postfix) with ESMTP	for <roland.news@gmx.at>; Tue, 15 May 2007 15:33:11 +0200 (CEST)
Subject: xxx
To: roland.news@gmx.at
Message-ID: <OF65B29854.5E0009EA-ONC12572DC.0047FA19@mdcs.at>
From: xxx.xxx@xx.at
Date: Tue, 15 May 2007 15:30:20 +0200
MIME-Version: 1.0
Content-type: multipart/mixed; 	Boundary="0__=4EBBF84FDFD47C898f9e8a93df938690918c4EBBF84FDFD47C89"
X-GMX-Antivirus: -1 (not scanned, may not use virus scanner)
X-GMX-Antispam: 0 (Mail was not recognized as spam)
X-GMX-UID: /McgKek9Mydy2KOnwWplMt9raGRhZloe

Migrated-From: http://trac.roundcube.net/ticket/1484423

**Comment by thomascube** [comment]
**Comment by thomascube on 13 Aug 2007 20:24 UTC**
Ticket #1484512 marked as duplicate of this bug.

**Comment by thomascube** [comment]
**Comment by imars on 14 Aug 2007 12:27 UTC**
As I had mentioned in duplicate #1484512, the messages I receive from my voicemail account exhibit similar behavior, as per attachment:"freedomvoice.txt". More precisely:

1. It is multipart/alternative, which contains an inline text/plain part in quoted-printable, which is not displayed, and then an audio/mpeg attachment, which is shown as available for download.

2. When we click on the MP3 attachment to save it locally, what we are sent is the contents of the text we should see displayed. There is no means to download the MP3.

3. In the message list as well as in the message view, the date is not displayed, even though it does appear in the headers. This may be unrelated; perhaps actually caused by the odd "To" field?

**Comment by thomascube** [comment]
**Comment by thomascube on 16 Aug 2007 14:49 UTC**
Must be some multipart parsing issue of the IMAP server. The message appears correctly on my testing installation.

**Comment by thomascube** [comment]
**Comment by imars on 16 Aug 2007 22:10 UTC**
I'm running Cyrus 2.2.12 which isn't exactly the most recent version... I'll upgrade the whole thing and report back to you.

**Comment by thomascube** [comment]
**Comment by imars on 16 Aug 2007 23:21 UTC**
I've successfully upgraded to Cyrus 2.3.8 and I still experience the two issues described: no date displayed in list or message views, and clicking on the MP3 downloads a file with the right name, but containing the plain text contents. I deleted the old test message and just sent myself a fresh voicemail now that I upgraded, just in case, with same results.

Maybe cclient or such a dependency on the PHP side is inadequate? Feel free to spy on my [phpinfo()](http://www.imars.com/~lis/info.php) in case there's something obvious. All it says about IMAP, though, is that it's using the "2004" version of c-Client. According to my sources (FreeBSD ports) 2004g seems to be the latest version.

I'm not too sure what to investigate next.

**Comment by thomascube** [comment]
**Comment by till on 1 Sep 2007 15:20 UTC**
First, upgrade to SVN.

Then enable the logs and let me know if you get any errors in there. Like anything.

If this doesn't show anything, please go in your index.php, line 73:

```
ini_set('error_reporting', E_ALL&~E_NOTICE);
```

Change to:
```
ini_set('error_reporting', E_ALL);
```

Check the logs again if anything '''relevant (to displaying your email)''' pops up.

**Comment by thomascube** [comment]
**Comment by imars on 11 Oct 2007 21:34 UTC**
Thanks for the procedure. Sorry for the delay, had a surge of overtime to deal with. We're moving hosting servers now, I'll get a SVN copy working by the end of the month and report back to you.

**Comment by thomascube** [comment]
**Comment by fraenki on 16 Oct 2007 11:26 UTC**
Replying to [roland.news@gmx.at](ticket:1484423):
> Some of my emails are shown without content. [...] The email can be shown without any probems in Outlook (using IMAP) and also with ilohamail (webmail), also using IMAP.

I can confirm this bug. RoundCube does not show any content, it only indicates an attachment (but there is none). The anonymized mail was attached to this ticket as "roundcube-mail.txt".

I am using FreeBSD 6.2 with Apache 2.2.6, PHP 5.2.4 and RoundCube 0.1.20070902 (a SVN snapshot inside the FreeBSD Ports Collection). I have also tried Linux with Apache 2.0.59, PHP 5.2.3 and RoundCube RC1.


Ciao

- Frank

**Comment by thomascube** [comment]
**Comment by cedricb on 26 Nov 2007 13:28 UTC**
> I can confirm this bug. RoundCube does not show any content, it only indicates an attachment (but there is none). The anonymized mail was attached to this ticket as "roundcube-mail.txt".

Hello, I confirm this behaviour too. Seen on some companies webmail.

Here is a short email source that does not work. Our emails are forged with javamail :

```
From - Thu Nov 22 10:14:09 2007
Return-Path: <xxxx@xxxxxxxx.xx>
Date: Thu, 22 Nov 2007 10:14:04 +0100
From: Test <xxxx@xxxxxxxx.xx>
Reply-To: Test <xxxx@xxxxxxxx.xx>
To: Receiver <yyyy@yyyyyyy.yy>
Subject: =?ISO-8859-15?Q?Test email?=
Mime-Version: 1.0
Content-Type: multipart/mixed; 
	boundary="----=_Part_47963_9821963.1195722844616"

------=_Part_47963_9821963.1195722844616
Content-Type: multipart/related; 
	boundary="----=_Part_47962_30260864.1195722844616"

------=_Part_47962_30260864.1195722844616
Content-Type: multipart/alternative; 
	boundary="----=_Part_47961_14158668.1195722844615"

------=_Part_47961_14158668.1195722844615
Content-Type: text/plain; charset="ISO-8859-1"
Content-Transfer-Encoding: quoted-printable


Example of an email that won't work with RoundCube.

=09=09=09
------=_Part_47961_14158668.1195722844615--

------=_Part_47962_30260864.1195722844616--

------=_Part_47963_9821963.1195722844616--

```


Kind Regards,
Cdric

**Comment by thomascube** [comment]
**Comment by till on 13 Feb 2008 01:48 UTC**
Can confirm all of them, looking into it again.

**Comment by thomascube** [comment]
**Comment by estadtherr on 13 Feb 2008 04:04 UTC**
It's weird that one of the parts is multipart/alternative, but there is only one alternative included. I wonder if the structure parsing bombs on this case...

**Comment by thomascube** [comment]
**Comment by till on 13 Feb 2008 12:57 UTC**
It's wrapped, wrapped.

**Comment by thomascube** [comment]
**Comment by till on 15 Feb 2008 00:22 UTC**
I debugged this call all night.

The structure parsing seems to be way off from Illohamail. I figured out that in ```rcmail_parse_message()``` no ```structure->parts``` are available, even though the email should have some. Well - in some cases at least 2 or 3.

Not sure if ". FETCH" fails, or a later parsing. This is not exactly a very readable response format. I guess we'll investigate over the weekend and if all fails postpone it to 0.1.1.

**Comment by thomascube** [comment]
**Comment by till on 15 Feb 2008 00:26 UTC**
```
use MIME::Parser;
$parser=new MIME::Parser;
$parser->output_under('.');
$parser->extract_uuencode(1);
$entity=$parser->parse(\*STDIN);
$entity->dump_skeleton;
```

Works better from perl. ;-) (Thanks to Trengo@freenode)

**Comment by thomascube** [comment]
**Comment by till on 16 Feb 2008 18:00 UTC**
I'm delaying this to 0.1.1.

Apparently this depends on the IMAP server used. Some IMAP servers return the "correct" structure - some don't. One of the examples is Courier-IMAP though I cannot rule out if this is a bug in my version and would be fixed if I update it.

A fix to correct this in RoundCube is in the works but it's too much to figure it out now.

**Comment by thomascube** [comment]
**Comment by alec on 15 May 2008 10:53 UTC**
Problem is also with dovecot 1.0.

**Comment by thomascube** [comment]
**Comment by alec on 15 Jun 2008 07:50 UTC**
Freedomvoice.txt and roundcube-mail.txt are working for me with latest svn-trunk. A month ago it wasn't working, so I assume it's fixed.

**Comment by thomascube** [comment]
**Comment by phs on 15 Jun 2008 19:02 UTC**
In Roundcube 0.2alpha it still seems to be broken. I attached the mail, that doesn't work for me in Roundcube 0.2alpha (see content_not_shown.txt and content_not_shown.png). Curiously it worked when I tried it with Roundcube 0.1rc2...

**Comment by thomascube** [comment]
**Comment by alec on 16 Jun 2008 09:54 UTC**
Replying to [phs](comment:18):
> In Roundcube 0.2alpha it still seems to be broken. I attached the mail, that doesn't work for me in Roundcube 0.2alpha (see content_not_shown.txt and content_not_shown.png). Curiously it worked when I tried it with Roundcube 0.1rc2...

That's another story, fixed in [5f8686e2]
